src/python/gudhi/reader_utils.cc

```diff
@@ -0,0 +1,113 @@
+ *    Author(s):       Vincent Rouvreau
+ *
+ *    Copyright (C) 2020 Inria
+ *
+ *    Modification(s):
+ *      - YYYY/MM Author: Description of the modification
+ */
+
+#include <gudhi/Points_off_io.h>
+#include <gudhi/reader_utils.h>
+
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+#include <pybind11/numpy.h>
+#include <pybind11/stl_bind.h>
+
+#if BOOST_VERSION < 106600
+# include <boost/function_output_iterator.hpp>  // for boost::make_function_output_iterator
+#else
+# include <boost/iterator/function_output_iterator.hpp>  // for boost::make_function_output_iterator
+#endif
+
+#include <vector>
+#include <array>
+#include <string>
+
+namespace py = pybind11;
+
+using Persistence_intervals = std::vector<std::array<double,2>>;
+PYBIND11_MAKE_OPAQUE(Persistence_intervals);
+
+PYBIND11_MODULE(reader_utils, m) {
+      m.attr("__license__") = "MIT";
+      m.def("read_points_from_off_file", [](const std::string& off_file) {
+            Gudhi::Points_off_reader<std::vector<double>> off_reader(off_file);
+            return off_reader.get_point_cloud();
+          },
+        py::arg("off_file"),
+        R"pbdoc(
+    Read points from OFF file.
+
+    :param off_file: Path to an `OFF <fileformats.html#off-file-format>`_ file.
+    :type off_file: string
+
+    :returns:  The point set.
+    :rtype: List[List[float]]
+        )pbdoc"
+      );
+      m.def("read_lower_triangular_matrix_from_csv_file", &Gudhi::read_lower_triangular_matrix_from_csv_file<double>,
+        py::arg("csv_file"), py::arg("separator") = ';',
+        R"pbdoc(
+    Read lower triangular matrix from a CSV style file.
+
+    :param csv_file: Path to a CSV file.
+    :type csv_file: string
+    :param separator: The value separator in the CSV file. Default value is ';'
+    :type separator: char
+
+    :returns:  The lower triangular matrix.
+    :rtype: List[List[float]]
+        )pbdoc"
+      );
+      m.def("read_persistence_intervals_grouped_by_dimension", &Gudhi::read_persistence_intervals_grouped_by_dimension,
+        py::arg("persistence_file"),
+        R"pbdoc(
+    Reads a file containing persistence intervals.
+    Each line might contain 2, 3 or 4 values: [[field] dimension] birth death
+    The return value is a `dict(dim, list(tuple(birth, death)))`
+    where `dim` is an `int`, `birth` a `float`, and `death` a `float`.
+    Note: the function does not check that birth <= death.
+
+    :param persistence_file: Path to a `persistence diagram <fileformats.html#persistence-diagram>`_ file.
+    :type persistence_file: string
+
+    :returns:  The persistence pairs grouped by dimension.
+    :rtype: Dict[int, List[Tuple[float, float]]]
+        )pbdoc"
+      );
+      py::bind_vector<Persistence_intervals>(m, "PersistenceIntervals", py::buffer_protocol());
+      m.def("read_persistence_intervals_in_dimension", [](const std::string& persistence_file, int only_this_dim) {
+            std::vector<std::array<double,2>> ret;
```

Comment:
I think this should use the Persistence_intervals typedef, to confirm that it is using the same type we made opaque and bound.

Reply:
Oh yes, my fault. Done on 63552a11